packages/dds/tree/src/test/utils.ts

```diff
@@ -83,11 +86,18 @@ export class TestTreeProvider {
      * await trees.ensureSynchronized();
      * ```
      */
-    public static async create(trees = 0): Promise<ITestTreeProvider> {
+    public static async create(trees = 0, summarizeOnDemand = false): Promise<ITestTreeProvider> {
         const provider = new TestTreeProvider() as ITestTreeProvider;
         for (let i = 0; i < trees; i++) {
             await provider.createTree();
+            if (summarizeOnDemand && i === 0) {
+                const summarizer = await createSummarizer(provider, provider.containers[0]);
+                this.summarizer = async () => {
```

Comment:
I think this code would be a little more clear if `this.summarizer` was the result of `createSummarizer` (the 'summarizer' on line 94) rather than an async function. Then, the body of 'summarize()' can call 'summarizeNow' instead, and we can remove this little function altogether.

Reply:
Updated `this.summarizer` to be the result of `createSummarizer()`, and called in the `summarize` method